pkg/sql/exec/rowstovec_tmpl.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2018 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// {{/*
+// +build execgen_template
+//
+// This file is the execgen template for distinct.eg.go. It's formatted in a
+// special way, so it's both valid Go and a valid text/template input. This
+// permits editing this file with editor support.
+//
+// */}}
+
+package exec
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/apd"
+	"github.com/cockroachdb/cockroach/pkg/sql/exec/types"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
+)
+
+// {{/*
+
+// Dummy import to pull in "apd" package.
+var _ apd.Decimal
+
+const (
+	_SEMANTIC_TYPE = sqlbase.ColumnType_SemanticType(0)
+	_WIDTH         = int32(0)
+)
+
+type _GOTYPE interface{}
+
+// */}}
+
+// EncDatumRowsToColVec converts one column from EncDatumRows to a column
+// vector. columnIdx is the 0-based index of the column in the EncDatumRows.
+func EncDatumRowsToColVec(
+	rows sqlbase.EncDatumRows,
+	vec ColVec,
+	columnIdx int,
+	columnType *sqlbase.ColumnType,
+	alloc *sqlbase.DatumAlloc,
+) error {
+	nRows := uint16(len(rows))
+
+	// {{define "rowsToColVec"}}
```

Comment:
This is an interesting way of doing a template function - just embedding its body into the function you're calling it from. I'd really like to get better execgen support for the "actual template functions" that Angela and I were playing with - the ones that act both as Go functions and subtemplates - so it'll be easier to do stuff like this in the future and have both the template code and the generated code make sense. I think this is fine - though the OCD person in me would prefer to see the template function pulled out in the way functions are pulled out in e.g. the hash joiner.

Reply:
Pulled out the function